content/blog/customer-Q-and-A.md

```diff
@@ -0,0 +1,758 @@
+To do this, you need to know that such a workflow (completely container-native, of course) always runs in a pod, is accessible via a service, and receives its schemas, properties, etc., via configMaps, for example.
+
+In the **simplest case** (though not suitable for production), if the SonataFlow operator (or OpenShift Serverless Logic Operator) is installed, you can simply create a SonataFlow Custom Resource. Example: https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/cloud/operator/build-and-deploy-workflows.html#build-deploy-workflow
+
+https://sonataflow.org/serverlessworkflow/main
+https://sonataflow.org/serverlessworkflow/main/cloud/operator/deployment-profile.html
+
+What then happens (in the background, through the operator):
+
+- A build with a BuildConfig is started - the standard builder image and thus the standard runtime image can be adapted - by default these are maven/jdk and Quarkus images.
+- When the build is finished, the workflow is started and the corresponding service and even a route are created (the route, i.e., the expose, can be suppressed).
+- There is also a Development Profile - CAUTION, workflows in DevMode are not entered into the so-called "Data Index Service" and are therefore not visible in RHDH - you can test them, play with them, but not interact with them via the RHDH Orchestrator.
+  https://kiegroup.github.io/kogito-docs/serverlessworkflow/latest/cloud/operator/developing-workflows.html
+
+</details>
+
+---
+
+## Authentication & Security
+
+<details>
+<summary><strong>Q: How do you pass credentials to functions which are defined as OpenAPI spec?</strong></summary>
+
+**A:** The workflow is using a token which is defined on RHDH.
+This is supported by https://backstage.io/docs/auth/service-to-service-auth/#access-restrictions
+
+In RHDH configmap there is the same section:
+
+```yaml
+backend:
+  auth:
+    externalAccess:
+      - type: static
+        options:
+          token: ${BACKEND_SECRET}
+          subject: orchestrator
+```
+
+The value for BACKEND_SECRET is taken from a secret of RHDH, e.g. backstage-backend-auth-secret (if installed by the Orchestrator operator)
+
+In the workflow, the secret is referenced from the application.properties property:
+
+quarkus.rest-client.scaffolder_openapi_yaml.url=${RHDH_URL}
+quarkus.openapi-generator.scaffolder_openapi_yaml.auth.BearerToken.bearer-token=${SCAFFOLDER_BEARER_TOKEN}
+
+The first one points to RHDH_URL (should use the internal service and not the openshift route to avoid certificate issues)
+The second property points to the value taken from the BACKEND_SECRET.
```

Comment:
new a new line before the second part, otherwise they are rendered as 1 single line